oasislmf/pytools/common/event_stream.py

```diff
@@ -337,33 +338,37 @@ def read_event(self, stream_in, main_selector, stream_selector, mv, byte_mv, cur
         """
         event_id = 0
         item_id = 0
-
-        while True:
-            if valid_buff < PIPE_CAPACITY:
-                stream_selector.select()
-                len_read = stream_in.readinto1(mv[valid_buff:])
-                valid_buff += len_read
-
-                if len_read == 0:
-                    stream_selector.close()
-                    main_selector.unregister(stream_in)
-                    if event_id:
-                        self.item_exit()
-                        return event_id, cursor, valid_buff
-
-                    break
-            cursor, event_id, item_id, yield_event = self.read_buffer(byte_mv, cursor, valid_buff, event_id, item_id, file_idx=file_idx)
-
-            if yield_event:
-                if 2 * cursor > valid_buff:
+        try:
+            while True:
+                if valid_buff < PIPE_CAPACITY:
+                    stream_selector.select()
+                    len_read = stream_in.readinto1(mv[valid_buff:])
+                    valid_buff += len_read
+
+                    if len_read == 0:
+                        stream_selector.close()
+                        main_selector.unregister(stream_in)
+                        if event_id:
+                            self.item_exit()
+                            return event_id, cursor, valid_buff
+
+                        break
+                cursor, event_id, item_id, yield_event = self.read_buffer(byte_mv, cursor, valid_buff, event_id, item_id, file_idx=file_idx)
+
+                if yield_event:
+                    if 2 * cursor > valid_buff:
+                        mv[:valid_buff - cursor] = mv[cursor: valid_buff]
+                        valid_buff -= cursor
+                        cursor = 0
+                    return event_id, cursor, valid_buff
+                else:
                     mv[:valid_buff - cursor] = mv[cursor: valid_buff]
                     valid_buff -= cursor
                     cursor = 0
-                return event_id, cursor, valid_buff
-            else:
-                mv[:valid_buff - cursor] = mv[cursor: valid_buff]
-                valid_buff -= cursor
-                cursor = 0
+
+        except Exception as e:
+            self.logger.error(str(e))
+            raise OasisException("Error reading stream", e)
```

Comment:
I think we want a subclass of OasisException , OasisStreamException to be sure this is specific.
otherwise later on in the try except it means the loop inside pytool cannot raise OasisException